src/core/maps/grid_map.h

```diff
@@ -81,48 +86,59 @@ class GridMap {
     if (!has_cell(cell_coord))
       return _unvisited_cell->value();
 
-  /**
-   * Returns a constant cell reference by the point.
-   * \param cell_coord The point with coordinates of requied cell in Grid Map.
-   */
-  const Cell &cell(const DiscretePoint2D& cell_coord) const {
-    return _cells[cell_coord.y][cell_coord.x];
+    return _cells[ROW_IND(cell_coord)][COL_IND(cell_coord)]->value();
   }
 
   /**
-   * Returns a cell by the coordinates.
-   * \param x,y Coordinates of a cell in Grid Map in cartesian coordinates.
+   * Projects coordinates in meters onto a cell of the grid map.
+   * \return The cell point corresponding to given coordinates.
    */
-  const Cell &cell(int x, int y) const {
-      return _cells[x][y];
+  DiscretePoint2D world_to_cell(double x, double y) const {
+    #define METERS_TO_CELLS(var)   \
+      std::floor((var)/_m_per_cell)
+
+    return DiscretePoint2D(METERS_TO_CELLS(x), METERS_TO_CELLS(y));
+    #undef METERS_TO_CELLS
   }
 
+  /// Returnes the scale.
+  double cell_scale() const { return _m_per_cell; }
+
   /**
-   * Returns a cell reference by the point.
+   * Returns the bounds of the cell in the World.
    * \param cell_coord The point with coordinates of requied cell in the grid
    *                                                                       map.
    */
-  Cell &cell(const DiscretePoint2D& cell_coord) {
-    return _cells[cell_coord.y][cell_coord.x];
+  Rectangle world_cell_bounds(const DiscretePoint2D &cell_coord) {
+    Rectangle bounds;
+    bounds.bot   = cell_coord.y * _m_per_cell;
+    bounds.top   = bounds.bot + _m_per_cell;
+    bounds.left  = cell_coord.x * _m_per_cell;
+    bounds.right = bounds.left + _m_per_cell;
+    return bounds;
   }
 
-  /**
-   * Tests whether the point is contained in a map.
-   * \param cell_coord The point with coordinates of requied cell in the grid
-   *                                                                       map.
+  /*!
+   * Returns the column index in container -
```

Comment:
Better: "Returns the \f$x\f$ coordinate of the map center". The same is relevant for map_center_y's comment.